lib/classes/Swift/Transport/StreamBuffer.php

```diff
@@ -91,7 +91,9 @@ public function startTLS()
         // To support modern tls we allow explicit tls1.0, tls1.1, tls1.2
         // Ssl3 and older are not allowed because they are vulnerable
         // @TODO make tls arguments configurable
-        return stream_socket_enable_crypto($this->stream, true, STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT);
+	// To Support more php versions
+	$supportTls = PHP_VERSION > '5.6' ? STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT : STREAM_CRYPTO_METHOD_TLS_CLIENT;
```

Comment:
I don't think it's going to work. I would do something like:

```
if (!defined('STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT')) {
    define('STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT', 0);
}

```

Reply:
I have tried another way. 

```php
defined('STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT') || define('STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT', 3);
defined('STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT') || define('STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT', 0);
defined('STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT') || define('STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT', 0);
```

this will be successful